### PR TITLE
Feature/improve coverage score

### PR DESCRIPTION
**Pull request by Unathi-Skosana**
### Change log (since  #56)

* Modified workflows to headlessly run tests that may require pygame to be initialized.
* Added new unittests and rewrote some; coverage score is now 80%.
* Amalgamated TODO items inside source files into a markdown file.
* Added game screenshots.

### CODE REVIEW

File: TODO.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- [ ] Make cursor appearance indicate condition such as unable to place a gate
```
Comment (HuangJunye) [inline TODO.md]: Can you put these into GitHub issues and remove this file?

File: tests/test_circuit_grid.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    def test_handle_single_qubit_gate_inputs(self):
+        """
+        Test handling input placing single
+        qubit gates on circuit grid
+        """
+
+        # 0 X-------
+        # 1 Y------
+        # 2 Z------
+        # 3 H------
```
Comment (HuangJunye) [inline tests/test_circuit_grid.py]: Unify the number of -- after gates

File: tests/test_circuit_grid.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        self.assertEqual(self.grid.selected_column, 0)
+        self.assertEqual(self.grid.selected_wire, 0)
+
+    def test_get_selected_node_gate_part(self):
+        """
+        Test getting gate on currently selected node
+        """
+
+        node1 = CircuitGridNode(node_types.Y)
+        node2 = CircuitGridNode(node_types.X)
```
Comment (HuangJunye) [inline tests/test_circuit_grid.py]: Can you please change to node1=X, node2=Y, node3=Z? Just to make it more clear

File: tests/test_circuit_grid.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+            node_types.EMPTY, self.circuit_grid_model.get_node_gate_part(1, 0)
+        )
+        self.assertEqual(
+            node_types.EMPTY, self.circuit_grid_model.get_node_gate_part(2, 0)
+        )
+        self.assertEqual(
+            node_types.EMPTY, self.circuit_grid_model.get_node_gate_part(3, 0)
+        )
+
+    def test_handle_multi_gate_inputs(self):
```
Comment (HuangJunye) [inline tests/test_circuit_grid.py:194]: Can you also test for cases when the controlled node is below the wire. For example, above the wire of X and Y and below the wire for Z and H.

File: tests/test_input.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+from qpong.utils.scene import Scene
+from qpong.utils.ball import Ball
+from qpong.utils.input import Input
+
+from qpong.model import circuit_node_types as node_types
+
+from qpong.utils.parameters import WINDOW_SIZE
+
+
+class TestLevel(unittest.TestCase):
```
Comment (HuangJunye) [inline tests/test_input.py:36]: Add tests for joysticks for the future. Can you please open an issue to track that?

File: tests/test_resources.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
+#
+# Copyright 2019 the original author or authors.
```
Comment (HuangJunye) [inline tests/test_resources.py]: Can you change the year in copyright from 2019 to 2022 for all files?

File: tests/test_scene.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    def test_start_scene(self):
+        """
+        Test start scene
+        """
+
+        # select easy mode
+        self.inject_event(pygame.KEYDOWN, key=pygame.K_a)
+        self.scene.start(self.screen, self.ball)
+        self.assertEqual(self.ball.initial_speed_factor, EASY)
+
```
Comment (HuangJunye) [inline tests/test_scene.py:84]: add a comment `select normal mode`

File: tests/test_scene.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        self.inject_event(pygame.KEYDOWN, key=pygame.K_b)
+        self.scene.start(self.screen, self.ball)
+        self.assertEqual(self.ball.initial_speed_factor, NORMAL)
+
```
Comment (HuangJunye) [inline tests/test_scene.py:89]: similar to above

**Review by HuangJunye — COMMENTED** [review]
Great work! Just a few simple comments

File: TODO.md (hunk 1 above)
Comment (Unathi-Skosana) [reply TODO.md]: Removed file, will open as issues soon.

File: tests/test_circuit_grid.py (hunk 2 above)
Comment (Unathi-Skosana) [reply tests/test_circuit_grid.py]: The number of ascii dashes on each ascii diagram is now 9.

File: tests/test_circuit_grid.py (hunk 4 above)
Comment (Unathi-Skosana) [reply tests/test_circuit_grid.py:194]: Y and Z are on the second wire now, and controlled from below. I did this instead as pylint was emitting a duplicate-code message.

**Comment by HuangJunye** [comment]
Looks great. Thanks a lot for working on it!

**Pull request merged**